### DISCUSSION TITLE
[Manipular xlsx]

### DISCUSSION THREAD

**Discussion by Ludsonfrancisco** [frontendbr/forum#2309] [Perguntas e Respostas]
Boa noite a todos, gostaria de uma ajuda.
Estou trabalhando em um projeto cujo a ideia é a seguinte: 
Criar um input file no html aonde eu consiga enviar uma planilha em xlsx, receberia essa planilha no meu node,js, manipularia ela  e retornaria alguns calculos no front, em uma outra pagina HTML, tipo uma tabela dinamica.
Obs. como é algo para automatizar no meu trabalho, eu ja saberia as colunas que receberia dessa planilha, entao já deixaria o algoritimo pronto. 

Estou estagnado no multer, fui até a parte de receber o arquivo usando multer, mas na hora de unir esse arquivo recebido com o codigo feito usando o package XLSX, travei.  Gostaria de algumas dicas e luz.

**Comment by felquis** [comment]
Olá, @Ludsonfrancisco tudo bem?

1. Sua dúvida não está na parte de enviar o arquivo xlsx para analisar, mas sim na parte de ler o conteúdo deste arquivo.
2. Sua dúvida não parece estar em ler o conteúdo do xlsx E manipular as colunas devolvendo uma nova planilha manipulada. Mas sim, devolvendo um HTML com os dados analisados.

Você já tentou usar algo como isso? https://www.npmjs.com/package/xlsx 

Doc: https://docs.sheetjs.com/docs/installation/nodejs
ex: https://github.com/SheetJS/sheetjs/wiki/Reading-XLSX-from-FileReader.readAsArrayBuffer()

EDIT: a doc parece ser muito bem feita https://docs.sheetjs.com/docs/solutions/input#example-user-submissions este exemplo me deixou achando que preciso testar essa lib.

---

Recentemente precisei apenas ler uma planilha em formato CSV, e inserir algumas linhas no banco, e achei os packages disponíveis bem complicadinhos de entender, e como minha planilha era muito simples, sem formulas e no formado csv. Consegui escrever rapidamente um parser que deixou o programa na minha opinião levemente mais são. Então se vc puder fazer isso eu até recomendo. A não ser que encontre um package que faça sentido para seu caso de uso.

---

Enfim, é isso que posso te dizer agora, qualquer novidade manda mensagem aqui para eu acompanhar, abraço boa sorte.
